wrapanapi/rhevm.py

```diff
@@ -553,3 +535,199 @@ def usage_and_quota(self):
+        disk_attachments_service = self._get_disk_attachments_service(vm_name)
+        for disk_attachment_service in disk_attachments_service.list():
+            disk = self.api.follow_link(disk_attachment_service.disk)
+            if disk.name == disk_name:
+                return disk_attachments_service.service(disk.id)
+        raise ItemNotFound(disk_name, 'disk')
+
+    def is_disk_attached_to_vm(self, vm_name, disk_name):
+        try:
+            return bool(self._get_disk_attachment_service(vm_name, disk_name))
+        except ItemNotFound:
+            return False
+
+    def get_vm_disks_count(self, vm_name):
+        return len(self._get_disk_attachments_service(vm_name).list())
+
+    def _get_disk_service(self, disk_name):
+        disks_service = self.api.system_service().disks_service()
+        query_result = disks_service.list(search="name={}".format(disk_name))
+        if not query_result:
+            raise ItemNotFound(disk_name, 'disk')
+        else:
+            disk = query_result[0]
+            return disks_service.service(disk.id)
+
+    def does_disk_exist(self, disk_name):
+        try:
+            return bool(self._get_disk_service(disk_name))
+        except ItemNotFound:
+            return False
+
+    def _check_disk(self, disk_id):
+        disks_service = self.api.system_service().disks_service()
+        disk_service = disks_service.disk_service()
+        disk = disk_service.get()
+        return disk.status == types.DiskStatus.OK
+
+    def add_disk_to_vm(self, vm_name, storage_domain=None, size=None, interface=None, format=None,
+            active=True):
+        disk_attachments_service = self._get_disk_attachments_service(vm_name)
+        disk_attachment = disk_attachments_service.add(
+            types.DiskAttachment(
+                disk=types.Disk(
+                    format=format,
+                    provisioned_size=size,
+                    storage_domains=[
+                        types.StorageDomain(
+                            name=storage_domain,
+                        )
+                    ]
+                ),
+                interface=interface,
+                active=active
+            )
+        )
+        wait_for(self._check_disk, func_args=[disk_attachment.disk.id], delay=5, num_sec=900,
+                 message="check if disk is attached")
+
+    def connect_direct_lun_to_appliance(self, vm_name, disconnect):
+        """Connects or disconnects the direct lun disk to an appliance.
+
+        Args:
+            vm_name: Name of the VM with the appliance.
+            disconnect: If False, it will connect, otherwise it will disconnect
+        """
+        if 'provider_key' in self.kwargs:
+            provider_name = self.kwargs['provider_key']
+        else:
+            raise TypeError("provider_key not supplied to the provider.")
+        # check for direct lun definition on provider's cfme_data.yaml
```

Comment:
I would prefer if we kept the LUN name as an explicitly defined arg on this method, rather than something that gets looked up via self.kwargs. In other words, IMO we shouldn't need to know about the format of a 'cfme_data' config in order to use wrapanapi.

Reply:
Looking at this method again, so I guess we'll need `lun_name`, `lun_ip_addr`, `lun_port`, `lun_icsi_target` as kwargs